framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -353,13 +372,54 @@ public static AnnotatedTypeMirror asMemberOf(
                 return atypeFactory.fromElement(elem);
             default:
                 AnnotatedTypeMirror type = asMemberOfImpl(types, atypeFactory, t, elem);
+                type = asMemberOfImpl(types, atypeFactory, t, elem, type);
+
                 if (!ElementUtils.isStatic(elem)) {
                     atypeFactory.postAsMemberOf(type, t, elem);
                 }
                 return type;
         }
     }
 
+    public static AnnotatedTypeMirror asMemberOfPreSubstitution(
```

Comment:
Can you add javadoc to the modified methods to make it clear what they are doing.